src/components/application_manager/test/commands/mobile/register_app_interface_test.cc

```diff
@@ -0,0 +1,158 @@
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "utils/shared_ptr.h"
+
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/commands/command_request_test.h"
+#include "application_manager/mock_policy_handler_interface.h"
+#include "connection_handler/mock_connection_handler.h"
+
+#include "mobile/register_app_interface_response.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+using namespace application_manager;
+
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::_;
+
+class RegisterAppInterfaceResponceTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  void SetUp() OVERRIDE {
+    message_ = utils::MakeShared<SmartObject>(::smart_objects::SmartType_Map);
+    (*message_)[strings::msg_params] =
+        ::smart_objects::SmartObject(::smart_objects::SmartType_Map);
+    command_sptr_ = CreateCommand<
+        application_manager::commands::RegisterAppInterfaceResponse>(
+        CommandsTest::kDefaultTimeout_, message_);
+  }
+
+  MessageSharedPtr message_;
+  utils::SharedPtr<commands::RegisterAppInterfaceResponse> command_sptr_;
+  policy_test::MockPolicyHandlerInterface mock_policy_handler_;
+};
+
+TEST_F(RegisterAppInterfaceResponceTest,
+       Run_ApplicationRegisteredAlready_Unsuccess) {
+  (*message_)[strings::msg_params][strings::success] = false;
+  (*message_)[strings::msg_params][strings::result_code] =
+      mobile_apis::Result::APPLICATION_REGISTERED_ALREADY;
+
+  EXPECT_CALL(mock_app_manager_,
+              SendMessageToMobile(
+                  MobileResultCodeIs(
+                      mobile_apis::Result::APPLICATION_REGISTERED_ALREADY),
+                  false)).Times(1);
+
+  EXPECT_CALL(mock_app_manager_, application(_)).Times(0);
+  command_sptr_->Run();
+}
+
+TEST_F(RegisterAppInterfaceResponceTest, Run_WrongConnectionKey_Unsuccess) {
+  EXPECT_CALL(mock_app_manager_, SendMessageToMobile(_, _)).Times(1);
+
+  ApplicationSharedPtr null_application;
+  EXPECT_CALL(mock_app_manager_, application(_))
+      .WillOnce(Return(null_application));
+
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler()).Times(0);
+  command_sptr_->Run();
+}
+
+TEST_F(RegisterAppInterfaceResponceTest,
+       Run_PolicyEnabled_SetHeartbeatTimeout) {
+  const std::string kPolicyAppId = "id";
```

Comment:
@Kozoriz, `kPolicyAppId` defined in both tests. Please move it to nameless namespace and check everywhere on possibility to do that.